SC4Cartographer/SC4Cartographer/MapApperance/MapApperanceSerializer.cs

```diff
@@ -0,0 +1,184 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.IO;
+using System.Drawing;
+
+
+namespace SC4CartographerUI
+{
+    public class MapApperanceSerializer
+    {
+        private const string tempFilename = "map_appearance_autosave.sc4cart";
+        private readonly string tempFilePath = Path.Combine(Path.GetTempPath(), tempFilename);
+        private const int VERSION = 1;
+
+        public void SaveToUserTempFolder(MapCreationParameters parameters)
+        {
+            SaveToFile(parameters, tempFilePath);
+        }
+
+        public void TryLoadFromUserTempFolder(MapCreationParameters parameters)
+        {
+            if (File.Exists(tempFilePath))
+            {
+                try
+                {
+                    LoadFromFile(parameters, tempFilePath);
+                }
+                catch (Exception)
+                {
+                    // should probably log this
+                }
+            }
+        }
+
+        /// <summary>
+        /// Save the a MapParameters object to a file.
+        /// Does not handle exceptions.
+        /// </summary>
+        /// <param name="parameters">The MapParameters object to save.</param>
+        /// <param name="path">Path to file to save to</param>
+        public void SaveToFile(MapCreationParameters parameters, string path)
+        {
+            List<string> properties = parameters.ToStrings();
+            properties.Insert(0, $"Version:{VERSION};");
+            properties.Insert(1, "!!!WARNING: This file is Case-Sensitive!!!");
+
+            WritePropertiesToFile(path, properties);
+        }
+
+        private static void WritePropertiesToFile(string path, List<string> properties)
```

Comment:
Personally I find smaller methods with single responsibilities, and on the same abstraction level, more readable. if you disagree, we'll go with the larger one. but honestly I'd take this a step further and separate it out totally:

```
        public void SaveToFile(MapCreationParameters parameters, string path)
        {
            var properties = PropertiesToStringList(parameters);
            WritePropertiesToFile(path, properties);
        }
```

Reply:
I don't find that more readable, it means that I have to look through 3 methods to work out how the `MapCreationParameters` are saved to a file instead of 1. It isn't even consistent with other methods in the file, such as `LoadFromFile` which has all the functionality easily assessible in one place.

I think also the seperated out functionality is fairly basic and only used in one flow, I don't think they warrent their own methods outside of personnal stylistic preference. If these descrete sets of functionality were going to be used in different places then I think there would be more merit to seperating out.

But as it stands this is overly fragmented code to me, that has only been seperated out based on subjective preference for readabilty, which I don't believe is enough of a reason for it to be laid out like this.